unlock-app/src/services/web3Service.js

```diff
@@ -279,11 +152,7 @@ export default class Web3Service extends EventEmitter {
       },
       (error, events = []) => {
         events.forEach(event => {
-          const transaction = {
-            hash: event.transactionHash,
-          }
-          this.emit('transaction.new', transaction)
-          this.getTransaction(transaction.hash)
+          this.emit('transaction.new', event.transactionHash)
```

Comment:
After the dust settles, it might worthwhile for to extract this method for uniform handling of events. `getPastUnlockTransactionsForUser` & ` getPastLockTransactions` are doing the same thing here

Reply:
You are right there is little difference between the 2. I'll refactor in a later PR1